src/pylero/session.py

```diff
@@ -131,7 +131,10 @@ def __init__(self, server, timeout):
     def _login(self):
         """login to the Polarion API"""
         sc = self._session_client
-        sc.service.logIn(self._server.login, self._server.password)
+        if self._server.token:
+            sc.service.logInWithToken("AccessToken", None, self._server.token)
```

Comment:
- did you try providing `username` here?
``` xml
<element name="logInWithToken">
 <complexType>
  <sequence>
   <element name="mechanism" type="xsd:string"/>
   <element name="username" type="xsd:string"/>
   <element name="token" type="xsd:string"/>
  </sequence>
 </complexType>
</element>
```
- just curious if we provide username here, maybe we'll be able to get that back with `security.service.getUserFromToken` 🤔 
- if that works we can change the code according to that or else PR lgtm.

Reply:
According to:
https://almdemo.polarion.com/polarion/sdk/doc/javadoc/com/polarion/alm/ws/client/session/SessionWebService.html#logInWithToken(java.lang.String,java.lang.String,java.lang.String)

username - the name of the user to login. Required for TcSS. null for AccessToken

that's why use None here. I'll try and let you know what'll happen.

And for access token, we are not supposed to require user provide the username.

Reply:
Based on javadoc no need to try.